Framework/Algorithms/test/CorelliPowderCalibrationDatabaseTest.h

```diff
@@ -0,0 +1,431 @@
+
+    // Output 2: search the saved output calibration file
+    boost::filesystem::path pdir(calibdir);
+    boost::filesystem::path pbase("corelli_instrument_20201117.csv");
+    boost::filesystem::path ptodaycalfile = pdir / pbase;
+    std::string todaycalfile = ptodaycalfile.string();
+    TS_ASSERT(boost::filesystem::exists(todaycalfile));
+    // load and compare
+    // ... ...
+
+    // Output 1: check all the files
+    std::vector<std::string> compnames{"source", "sample", "bank1", "bank2",
+                                       "bank42"};
+    std::vector<size_t> expectedrows{2, 2, 1, 1, 1};
+    for (size_t i = 0; i < 5; ++i) {
+      verify_component_files(calibdir, compnames[i], expectedrows[i]);
+    }
+
+    // Clean memory
+    // Remove workspace from the data service.
+    // AnalysisDataService::Instance().remove(outWSName);
+  }
+
+private:
+  /**
+   * @brief Create testing CORELLI event workspace
+   * @return
+   */
+  EventWorkspace_sptr createTestEventWorkspace() {
+    // Name of the output workspace.
+    std::string outWSName("CorelliPowderCalibrationDatabaseTest_matrixWS");
+
+    IAlgorithm_sptr lei =
+        AlgorithmFactory::Instance().create("LoadEmptyInstrument", 1);
+    lei->initialize();
+    lei->setPropertyValue("Filename", "CORELLI_Definition.xml");
+    lei->setPropertyValue("OutputWorkspace",
+                          "CorelliPowderCalibrationDatabaseTest_OutputWS");
+    lei->setPropertyValue("MakeEventWorkspace", "1");
+    lei->execute();
+
+    EventWorkspace_sptr ws;
+    ws = AnalysisDataService::Instance().retrieveWS<EventWorkspace>(
+        "CorelliPowderCalibrationDatabaseTest_OutputWS");
+
+    // Add property start_time
+    ws->mutableRun().addProperty<std::string>("start_time",
+                                              "2020-11-17T12:57:17", "", true);
+
+    return ws;
+  }
+
+  /**
+   * @brief Create Test Calibration TableWorkspace
+   * @param outWSName
+   * @return
+   */
+  TableWorkspace_sptr
+  createTestCalibrationTableWorkspace(const std::string &outWSName) {
+
+    ITableWorkspace_sptr itablews = WorkspaceFactory::Instance().createTable();
+    AnalysisDataService::Instance().addOrReplace(outWSName, itablews);
+
+    TableWorkspace_sptr tablews =
+        std::dynamic_pointer_cast<TableWorkspace>(itablews);
+    TS_ASSERT(tablews);
+
+    // Set up columns
+    for (size_t i = 0;
+         i < CorelliCalibration::calibrationTableColumnNames.size(); ++i) {
+      std::string colname = CorelliCalibration::calibrationTableColumnNames[i];
+      std::string type = CorelliCalibration::calibrationTableColumnTypes[i];
+      tablews->addColumn(type, colname);
+    }
+
+    std::cout << "[DEBUG 0] Table workspace rows: " << tablews->rowCount()
+              << "\n";
+
+    // append rows
+    Mantid::API::TableRow sourceRow = tablews->appendRow();
+    sourceRow << "source" << 0. << 0. << -15.560 << 0. << 0. << 0. << 0.;
```

Comment:
to be consistent, can you use CORELLI's component names for source and sample ("moderator" and "sample-position")?